lets-encrypt/activate-ssl.sh

```diff
@@ -202,99 +202,5 @@ then
     sed -i "s|</FilesMatch.*|#|g" "$ssl_conf"
 fi
 
-# Methods
-# https://certbot.eff.org/docs/using.html#certbot-command-line-options
-default_le="--rsa-key-size 4096 --renew-by-default --no-eff-email --agree-tos --uir --hsts --server https://acme-v02.api.letsencrypt.org/directory -d $domain"
-
-standalone() {
-# Generate certs
-if eval "certbot certonly --standalone --pre-hook 'service apache2 stop' --post-hook 'service apache2 start' $default_le"
-then
-    echo "success" > /tmp/le_test
-else
-    echo "fail" > /tmp/le_test
-fi
-}
-tls-alpn-01() {
-if eval "certbot certonly --preferred-challenges tls-alpn-01 $default_le"
-then
-    echo "success" > /tmp/le_test
-else
-    echo "fail" > /tmp/le_test
-fi
-}
-dns() {
-if eval "certbot certonly --manual --manual-public-ip-logging-ok --preferred-challenges dns $default_le"
-then
-    echo "success" > /tmp/le_test
-else
-    echo "fail" > /tmp/le_test
-fi
-}
-
-methods=(standalone dns)
-
-create_config() {
-# $1 = method
-local method="$1"
-# Check if $CERTFILES exists
-if [ -d "$CERTFILES" ]
- then
-    # Generate DHparams chifer
-    if [ ! -f "$DHPARAMS" ]
-    then
-        openssl dhparam -dsaparam -out "$DHPARAMS" 4096
-    fi
-    # Activate new config
-    check_command bash "$SCRIPTS/test-new-config.sh" "$domain.conf"
-    exit
-fi
-}
-
-attempts_left() {
-local method="$1"
-if [ "$method" == "standalone" ]
-then
-    printf "%b" "${ICyan}It seems like no certs were generated, we will do 1 more try.\n${Color_Off}"
-    any_key "Press any key to continue..."
-#elif [ "$method" == "tls-alpn-01" ]
-#then
-#    printf "%b" "${ICyan}It seems like no certs were generated, we will do 1 more try.\n${Color_Off}"
-#    any_key "Press any key to continue..."
-elif [ "$method" == "dns" ]
-then
-    printf "%b" "${IRed}It seems like no certs were generated, please check your DNS and try again.\n${Color_Off}"
-    any_key "Press any key to continue..."
-fi
-}
-
-# Generate the cert
-for f in "${methods[@]}"; do "$f"
-if [ "$(grep 'success' /tmp/le_test)" == 'success' ]; then
-    rm -f /tmp/le_test
-    create_config "$f"
-else
-    rm -f /tmp/le_test
-    attempts_left "$f"
-fi
-done
-
-# Failed
-msg_box "Sorry, last try failed as well. :/
-
-The script is located in $SCRIPTS/activate-ssl.sh
-Please try to run it again some other time with other settings.
-
-There are different configs you can try in Let's Encrypt's user guide:
-https://letsencrypt.readthedocs.org/en/latest/index.html
-Please check the guide for further information on how to enable SSL.
-
-This script is developed on GitHub, feel free to contribute:
-https://github.com/nextcloud/vm
-
-The script will now do some cleanup and revert the settings."
-
-# Cleanup
-apt remove certbot -y
-apt autoremove -y
-clear
```

Comment:
In my opinion, everything removed here should be put into a function instead "generate_cert" or whatever. 

Right now it seems like you divided everything into smaller pieces. Is there any thought behind that decision? Why is it better?

Reply:
Yeah, It could be to implement if they only use for getting TLS. I just think maybe it looks simple and